src/main/java/ca/on/oicr/gsi/status/BasePage.java

```diff
@@ -2,113 +2,119 @@
-
-			server.navigation().forEach(link -> {
-				try {
-					link.append(writer);
-				} catch (final XMLStreamException e) {
-					throw new RuntimeException(e);
-				}
-			});
-			writer.writeStartElement("a");
-			writer.writeAttribute("href", server.documentationUrl());
-			writer.writeCharacters("API Docs");
-			writer.writeEndElement();
-
-			writer.writeEndElement();
-
-			writer.writeStartElement("div");
-			renderContent(writer);
-			writer.writeEndElement();
-			writer.writeEndElement();
-
-			writer.writeEndElement();
-			writer.writeEndDocument();
-		} catch (final XMLStreamException e) {
-			throw new RuntimeException(e);
-		}
-	}
+  private static final Header CSS =
+      Header.css(
+          "body { margin:0; font-family:sans-serif; }\n" //
+              + "h1 { font-size:120%; }\n" //
+              + "body > div { margin:1em; }\n"
+              + "nav { background-color:#222; display:block; margin:0; padding: 0.5em }\n" //
+              + "nav > a, nav > div > span { display:inline-block; padding:0.5em; text-decoration:none; font-weight:400; font-size:120% }\n" //
+              + "nav > a:link, nav > a:visited, nav > div > span { color:#9d9d9d; } \n" //
+              + "nav > a:hover, nav > a:active, nav > div:hover > span { color:#fff; }\n" //
+              + "nav > div { position: relative; display: inline-block; }\n"
+              + "nav > div:hover div { display: block; }\n" //
+              + "nav > div > div { position:absolute; min-width: 200px; background-color: #fff; border:1px solid #000; box-shadow: 0 6px 12px rgba(0,0,0,0.175); border-radius:4px; z-index:1000; display:none; }\n" //
+              + "nav > div > div a { display:block; text-decoration:none; font-weight:400; padding: 0.5em 1em }\n" //
+              + "nav > div > div a:first-child { border-top-left-radius: 4px; border-top-right-radius: 4px }\n" //
+              + "nav > div > div a:last-child { border-bottom-left-radius: 4px; border-bottom-right-radius: 4px }\n" //
+              + "nav > div > div a:link, nav > div > div a:visited, nav > div > div a:hover, nav > div > div a:active { color:#000; }\n" //
+              + "nav > div > div a:hover, nav > div > div a:active { background-color:#eee; }\n" //
+              + "nav > :first-child { font-weight: bold }\n" //
+              + "table { width:100%; border-spacing:0; border-collapse:collapse; border-color:grey; margin-top:1ex; }\n" //
+              + "td,th { border:none; }\n"
+              + "td { border-top:1px solid #ddd; }\n"
+              + ".even td { width:50%; }\n" //
+              + "th { font-weight:700; text-align:left; }\n"
+              + "tr:hover { background-color:#f5f5f5; }");
+  private final boolean autorefresh;
+  private final ServerConfig server;
+
+  public BasePage(ServerConfig server, boolean autorefresh) {
+    super();
+    this.server = server;
+    this.autorefresh = autorefresh;
+  }
+
+  /** The URL of this page */
+  public String activeUrl() {
```

Comment:
Please Javadoc to say what this should return exactly - a URL relative to the base webapp URL, and without a leading slash, I think? Shouldn't it always be overridden too? If so, make abstract. If not, explain in Javadoc

Reply:
Updated.